Allow autoscroll and auto-redirect for the submission progress

yanniboi opened the pull request.

Commit:
By yanniboi: Allow autoscroll and auto-redirect for the submission progress route.

yanniboi commented on modules/simplytest_submission/js/progress.js:
```diff
@@ -14,20 +14,39 @@
+      $( preEl ).css("width", '100%');
+      $( preEl ).css("max-height", '300px');
+      $( preEl ).css("overflow", 'auto');
+
```
This needs to go into a css file so need to add it to a library.

Commit:
By yanniboi: Code cleanup.

Commit:
By yanniboi: Added css file to progress library.

patrickd- commented:
The collapsible area and the autoscroll are awesome, i'm a little worried about the redirect though. It's ticket off by default and the submitter (if the submission is cached and done instantly) has to look at the log output to find an URL and then copy that URL into the address bar in order to get to the page because the checkbox does no longer work after the deploy process is done.

The easiest fix might be to add a dummy button that is disabled by default and has no functionality. If the deployment is done the javascript attaches a redirect function when the button is clicked and removes the disabled state. Or can you think of something better?

While this is all work-in-progress it's still actively used for demo-ing opensocial, that's why i'm a bit worried about UX here.

Besides that this is awesome though!

yanniboi commented:
Yup I agree.

My plan was to have a hidden button that appears when it is finished. The js as you described it would work well. I think ultimately we want to have all the output tracked and output in php rather than js because that then allows us to support js-disabled browsers, and allows us to do actual progress better (as well as saving the final url on the submission).

But for now that is the best action I think.

Commit:
By yanniboi: Updated progress bar percent and added redirect button to progress page.

yanniboi commented:
Ok update:

1. I have added some 'echo' commands to the tops of the different script stages. They now echo "Starting next script - Build Server" or similar.
2. This means that I can now user the progress.js to move the progress bar along when the different stages of the build start - Starting at 30%, adding 10% at every stage, moving to 100% on completion.
3. I have also added a button to the form which is hidden by js and is shown at the end.
4. I have also collapsed the debug by default, but it can still autoredirect if you expand it and check the box.

yanniboi commented:
![output-iloveimg-compressed 1](https://cloud.githubusercontent.com/assets/1744957/26031665/da4d9a58-3873-11e7-88ba-d01ffa9ceef0.gif)

patrickd- reviewed (approved):
:+1: Looking good :)
